Fixing peak area, rubberband, and M3 and CRISM io bugs

ryanbanderson opened the pull request.

jlaura commented on libpyhat/io/io_crism.py:
```diff
@@ -12,7 +12,7 @@ class Crism(GeoDataset, HCube):
     An Crism specific reader with the spectral mixin.
     """
-    def __init__(self, file_name):
+    def __init__(self, file_name, libpyhat_enabled = True):
```
Why is this passed in? and then never used or set to an instance attribute?

ryanbanderson replied:
The code already had the following in it, but was failing because libpyhat_enabled was not defined:

```
if libpyhat_enabled:
            self.derived_funcs = get_derived_funcs(crism)
```

ryanbanderson replied:
So alternatively we could just get rid of the if statement

ryanbanderson replied:
@tthatcher95 thoughts on this? do we need the if statement?

tthatcher95 replied:
@ryanbanderson We can probably just set the derived_funcs no matter what
@acpaquette Do you remember why we did this?

ryanbanderson replied:
@tthatcher95 Ok, that's what I was thinking. You want to make that change or should I?

tthatcher95 replied:
@ryanbanderson You can do it, you have the PR already open

ryanbanderson replied:
ok, done

ryanbanderson commented:
@jlaura do these changes look ok to you? Travis isn't triggering - not sure if it's waiting on approval or if there's some other issue...

jlaura commented:
Two things:

- Why does this PR have peak area and rubber band code in it?
- The merge conflicts need to be resolved.

ryanbanderson commented:
Conflicts resolved. I found and fixed the peak area and rubber band bugs while waiting for approval. I've updated this PR's name to reflect that they are included.

jlaura commented:
In the future these need to be new PRs. Just make a branch off dev and fix the bugs in their own branch. I can't review the new code because I don't know what it is supposed to do. Does the new code have tests? If not coveralls will fail.

jlaura commented:
@ryanbanderson Can you rebase off master and make sure this is up to date with dev. Travis is complaining about merge issues: https://travis-ci.org/github/USGS-Astrogeology/PyHAT/requests

ryanbanderson commented:
@jaluara Ok, I did the rebase. Looks like Travis is passing but not updating here. Any ideas?

jlaura commented:
@ryanbanderson If you go here: https://travis-ci.org/github/USGS-Astrogeology/PyHAT/pull_requests it is possible to check the status on the build even if a hook fails. Looks like this passed 19 hours ago.  One final question - why no tests to the rubber band and peak area code?

ryanbanderson commented:
Basically because I don't know how to write a test and I never budgeted for that extra work. There's a bunch of code from my side of the project that is not tested and it's too late to change that now. My project ends very soon and I need to get bugs fixed and the code installable asap. In the future, either under Itiya's project or IAA funding, I'm all for getting everything tested.

jlaura commented:
Test aren't extra work. They are the only way that you repeatedly know your code actually works. Please feel free to push those changes into a branch, but we can't release software that isn't being tested. This isn't my policy. This is USGS policy about code having to go through a review for release.

ryanbanderson commented:
Ok, I'll get some sample data and then @tthatcher95 is going to teach me to do tests and we'll try to crank through some, starting with rubberband and peak areas for this PR.

jlaura commented:
@ryanbanderson :+1: Thank you. If you need assistance please feel free to reach out me as well.

ryanbanderson commented:
@jlaura ok, I think I got tests working for rubberband and peak_area. How do we merge this PR if the travis hook is failing?

ryanbanderson commented:
Strange... now it looks like Travis is not even triggering with my new commits. I'm thinking we may need to close this PR and open a new one to get Travis to behave.
